ci/test.lua

```diff
@@ -116,29 +114,87 @@ end
 test_envvars.require = clean_require
 test_envvars.reqscript = clean_reqscript
 
+local function is_title_screen(scr)
+    scr = scr or dfhack.gui.getCurViewscreen()
+    return df.viewscreen_titlest:is_instance(scr)
+end
+
+-- This only handles pre-fortress-load screens. It will time out if the player
+-- has already loaded a fortress or is in any screen that can't get to the title
+-- screen by sending ESC keys.
 local function ensure_title_screen()
-    if df.viewscreen_titlest:is_instance(dfhack.gui.getCurViewscreen()) then
-        return
-    end
-    print('Looking for title screen...')
-    for i = 0, 100 do
+    for i = 1, 100 do
         local scr = dfhack.gui.getCurViewscreen()
-        if df.viewscreen_titlest:is_instance(scr) then
+        if is_title_screen(scr) then
             print('Found title screen')
-            break
-        else
+            return
+        end
+        scr:feed_key(df.interface_key.LEAVESCREEN)
+        delay(10)
+        if i % 10 == 0 then print('Looking for title screen...') end
+    end
+    qerror(string.format('Could not find title screen (timed out at %s)',
+                         dfhack.gui.getCurFocus(true)))
+end
+
+local function is_fortress(focus_string)
+    focus_string = focus_string or dfhack.gui.getCurFocus(true)
+    return focus_string == 'dwarfmode/Default'
+end
+
+-- Requires that a fortress game is already loaded or is ready to be loaded via
+-- the "Continue Playing" option in the title screen. Otherwise the function
+-- will time out and/or exit with error.
+local function ensure_fortress()
```

Comment:
Calling out to the `load-save` script might be a better solution here. It should work from the title screen. I'm not sure if it will be easy to distinguish between failures due to no matching save vs. no saves at all, for instance, but it should work.

Incidentally, does this actually look for `region1`, or just any save? Maybe the region folder name could be a parameter.

Reply:
It doesn't look specifically for `region1`, it just selects the first region in the list. I hadn't thought of using `load-save` - we could certainly add a parameter to the test harness to control which region is loaded (only used if one isn't loaded already, of course).

Reply:
added `--save_dir` parameter and changed to use `load-save` script to load